### DISCUSSION TITLE
How to retrieve automatically the list of the PlantUML themes?

### DISCUSSION THREAD

**Discussion by The-Lum** [plantuml/plantuml#652] [Q&A] [answered]
- How to retrieve automatically the list of the PlantUML themes?

_[Hints: A beginning is the `help themes` command...]_

**Comment by matthew16550** [comment]
Hi @The-Lum, could you provide some more details here.

e.g. Where are you planning to use the list / what sort of automation are you considering?

**Reply by The-Lum** [reply]
Hello @matthew16550,

My main goal is to use this list in order to generate a Theme Gallery.
That is just a beginning, I have not yet any script _[bash, makefile, Github Action]_ and perhaps a list on a JSON format must be necessary...

Then, _about my first hint_, I just open this Issue: 
- #654

Another hint will be perhaps to retrieve the list from this directory:
- https://github.com/plantuml/plantuml/tree/master/themes

Regards.

**Comment by matthew16550** [comment]
For this use case I think it would be simplest if you can list files from the directory.

**Reply by The-Lum** [reply]
But I would also like it to be automatic and also dependent on what it would have in a particular `plantuml.jar`, and not just the latest version in GitHub.

**Comment by matthew16550** [comment, marked as answer]
Maybe something like this:
`echo "@startuml\nhelp themes\n@enduml" | java -jar plantuml.jar -pipe -txt | awk '/^ {10,}[[:alnum:]]/ { print $1 }'`
